src/mattack_actors.cpp

```diff
@@ -291,8 +291,26 @@ mattack_actor *gun_actor::clone() const
     return new gun_actor( *this );
 }
 
+// @todo Enforce load order (add finalization?) and allow calculating this at json load
+int gun_actor::get_range() const
+{
+    if( target_range > 0 ) {
+        return target_range;
+    }
+
+    item gun( gun_type );
+    itype_id ammo = ( ammo_type != "NULL" ) ? ammo_type : gun.ammo_default();
+
+    if( ammo != "NULL" ) {
```

Comment:
gun.ammo_default() can return "NULL", and while it isn't going to crash, I think it's going to print a debugmsg in that case despite the fact that that is a valid  thing to do in this case.